File: fueltracker-android/app/src/main/java/com/example/fueltracker_android/domain/model/Stats.kt
package com.example.fueltracker_android.domain.model

object Stats {

    data class Summary(
        val totalSpent: Double,
        val totalLiters: Double,
        val avgConsumption: Double?,   // null если < 2 заправок
        val bestConsumption: Double?,
        val worstConsumption: Double?,
        val avgPricePerLiter: Double,
        val refuelsCount: Int,
        val totalKm: Int
    )

    data class MonthlyData(
        val month: String,
        val totalSpent: Double,
        val totalLiters: Double,
        val avgConsumption: Double?,   // null если нет данных расхода
        val avgPrice: Double,
        val refuelsCount: Int
    )

    data class ConsumptionPoint(
        val date: String,
        val consumption: Double?,      // null для первой заправки
        val odometer: Int
    )
}

File: fueltracker-backend/app/schemas/stats.py
from pydantic import BaseModel
from typing import Optional


# --- Сводная статистика пользователя ---

class SummaryResponse(BaseModel):
    """Общая статистика по всем заправкам текущего пользователя."""
    user_id: int
    total_spent: float
    total_liters: float
    avg_consumption: Optional[float]
    best_consumption: Optional[float]
    worst_consumption: Optional[float]
    avg_price_per_liter: float
    refuels_count: int
    total_km: int


# --- Месячная статистика ---

class MonthlyData(BaseModel):
    """Данные одного месяца."""
    month: str
    total_spent: float
    total_liters: float
    avg_consumption: Optional[float]
    avg_price: float
    refuels_count: int


class MonthlyResponse(BaseModel):
    """Месячная статистика текущего пользователя."""
    user_id: int
    data: list[MonthlyData]


# --- Тренды расхода топлива ---

class TrendPoint(BaseModel):
    """Одна точка на графике расхода топлива."""
    date: str
    consumption: Optional[float]
    odometer: int


class TrendResponse(BaseModel):
    """Тренд расхода топлива текущего пользователя."""
    user_id: int
    points: list[TrendPoint]